cedar-policy-validator/src/types.rs

```diff
@@ -2520,6 +2517,198 @@ mod test {
         .expect("Expected valid schema")
     }
 
+    #[test]
+    #[cfg(feature = "partial-eval")]
+    fn test_typecheck_partial_value_with_unknown() {
+        use cedar_policy_core::ast::Context;
+        let context = Context::from_json_value(serde_json::json!({
+            "a": {
+                "__extn": {
+                    "fn": "unknown",
+                    "arg": "test_arg"
+                }
+            },
+        }))
+        .unwrap();
+
+        let typeValidator = Type::record_with_attributes(
+            [
+                (
+                    "a".into(),
+                    AttributeType::optional_attribute(Type::primitive_boolean()),
+                ),
+                (
+                    "b".into(),
+                    AttributeType::optional_attribute(Type::primitive_boolean()),
+                ),
+                (
+                    "c".into(),
+                    AttributeType::optional_attribute(Type::primitive_boolean()),
+                ),
+            ],
+            OpenTag::ClosedAttributes,
+        );
+
+        let result = typeValidator
+            .typecheck_partial_value(&context.clone().into(), Extensions::all_available());
```

Comment:
All the tests use `Extensions::all_available()`, but what happens if we don't use any?